### PR TITLE
DNMY: Refactoring Storage 

### PR DESCRIPTION
**Pull request by sourabhdalvi**

### CODE REVIEW

File: src/descriptors/power_system_structs.json [hunk 1]
```diff
@@ -65,6 +65,69 @@
       ],
       "supertype": "OperationalCost"
     },
+    {
+        "struct_name": "StorageManagementCost",
```
Comment (jd-lara) [inline src/descriptors/power_system_structs.json:69]: Not sure if StorageManagementCost is the best descriptor here.

File: src/models/generated/BatteryEMS.jl [hunk 2]
```diff
@@ -17,9 +17,8 @@ This file is auto-generated. Do not edit.
         reactive_power::Float64
         reactive_power_limits::Union{Nothing, Min_Max}
         base_power::Float64
+        operation_cost::Union{Nothing, OperationalCost}
```
Comment (jd-lara) [inline src/models/generated/BatteryEMS.jl:20]: this will be a breaking API change. we need to add a constructor that makes the conversion correctly

**Review by jd-lara — COMMENTED** [review]
This PR makes a lot of API changes in the Storage objects, it will force a 2.0 release. We need to evaluate how to better address these changes.

File: src/deprecated.jl [hunk 3]
```diff
@@ -0,0 +1,24 @@
+
+# BEGIN 1.2.1  deprecations 
+@deprecate get_energy_value(val::BatteryEMS) get_energy_surplus_cost(
```
Comment (jd-lara) [inline src/deprecated.jl:3]: We need a test for this. Please add a new test file for deprecations. `test_deprecations.jl`

Comment (sourabhdalvi) [reply src/deprecated.jl:3]: added the test file

**Review by jd-lara — APPROVED** [review]
The PR looks good to me but we can't merge until we fix the breaking change for PSI. We need to wait to merge.

File: src/descriptors/power_system_structs.json (hunk 1 above)
Comment (jd-lara) [reply src/descriptors/power_system_structs.json:69]: @claytonpbarrows what do you think about this name?

Comment (claytonpbarrows) [reply src/descriptors/power_system_structs.json:69]: I don't hate `StorageManagementCost`. But what about `StorageOperationalCost`?

**Pull request merged**